src/uu/ls/src/ls.rs

```diff
@@ -1200,18 +1249,18 @@ fn display_items(items: &[PathBuf], strip: Option<&Path>, config: &Config, comma
             max_size = size.max(max_size);
         }
         for item in items {
-            display_item_long(item, strip, max_links, max_size, config, command_line);
+            display_item_long(item, strip, max_links, max_size, config);
         }
     } else {
         let names = items.iter().filter_map(|i| {
-            let md = get_metadata(i, false);
+            let md = i.md.as_ref();
             match md {
                 Err(e) => {
-                    let filename = get_file_name(i, strip);
+                    let filename = get_file_name(&i.p_buf, strip);
```

Comment:
Do you think this could just be
```rust
let filename = i.file_name;
```
or is it too hard to implement `strip` then?

Reply:
Another nice suggestion 👍 
Actually, I think all get_file_name could be updated to be something like (based what all current use-cases look like)

```
fn get_file_name(p: &PathData, strip: bool) -> String {
    if strip {
        p.file_name
    } else {
        p.lossy_string
    }
}
```